src/main/java/org/eclipse/uprotocol/mqtt/HiveMqMQTT5Client.java

```diff
@@ -0,0 +1,288 @@
+ * terms of the Apache License Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.eclipse.uprotocol.mqtt;
+
+import com.google.protobuf.ByteString;
+import com.hivemq.client.mqtt.datatypes.MqttTopic;
+import com.hivemq.client.mqtt.datatypes.MqttTopicFilter;
+import com.hivemq.client.mqtt.mqtt5.Mqtt5AsyncClient;
+import com.hivemq.client.mqtt.mqtt5.Mqtt5Client;
+import com.hivemq.client.mqtt.mqtt5.datatypes.Mqtt5UserProperties;
+import com.hivemq.client.mqtt.mqtt5.message.publish.Mqtt5Publish;
+import com.hivemq.client.mqtt.mqtt5.message.publish.Mqtt5PublishBuilder;
+import com.hivemq.client.mqtt.mqtt5.message.publish.Mqtt5PublishResult;
+import org.eclipse.uprotocol.transport.UListener;
+import org.eclipse.uprotocol.transport.UTransport;
+import org.eclipse.uprotocol.uri.serializer.UriSerializer;
+import org.eclipse.uprotocol.uuid.serializer.UuidSerializer;
+import org.eclipse.uprotocol.v1.*;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.stream.Collectors;
+
+import static org.eclipse.uprotocol.v1.UCode.INTERNAL;
+import static org.eclipse.uprotocol.v1.UCode.OK;
+
+class HiveMqMQTT5Client implements UTransport {
+
+    private static final Logger LOG = LoggerFactory.getLogger(HiveMqMQTT5Client.class);
+    private static final String USER_PROPERTIES_KEY_FOR_ID = "1";
+    private static final String USER_PROPERTIES_KEY_FOR_MESSAGE_TYPE = "2";
+    private static final String USER_PROPERTIES_KEY_FOR_SOURCE_NAME = "3";
+    private static final String USER_PROPERTIES_KEY_FOR_SINK_NAME = "4";
+    private static final String USER_PROPERTIES_KEY_FOR_PRIORITY = "5";
+    private static final String USER_PROPERTIES_KEY_FOR_TTL = "6";
+    private static final String USER_PROPERTIES_KEY_FOR_PERMISSION_LEVEL = "7";
+    private static final String USER_PROPERTIES_KEY_FOR_COMMSTATUS = "8";
+    private static final String USER_PROPERTIES_KEY_FOR_REQID = "9";
+    private static final String USER_PROPERTIES_KEY_FOR_TOKEN = "10";
+    private static final String USER_PROPERTIES_KEY_FOR_TRACEPARENT = "11";
+    private static final String USER_PROPERTIES_KEY_FOR_PAYLOAD_FORMAT = "12";
+    private final Mqtt5AsyncClient client;
+    private final UUri source;
+
+    public HiveMqMQTT5Client(UUri source, Mqtt5Client client) {
+        this.client = client.toAsync();
+        this.source = source;
+    }
+
+    @Override
+    public CompletionStage<UStatus> send(UMessage uMessage) {
+        LOG.trace("should send a message:\n{}", uMessage);
+        CompletableFuture<UStatus> result = new CompletableFuture<>();
+
+        Mqtt5UserProperties userProperties = Mqtt5UserProperties.builder()
```

Comment:
user properties should only be added if the uAttribute value is defined. https://github.com/eclipse-uprotocol/up-spec/blob/main/up-l1/mqtt_5.adoc#:~:text=Only%20UAttributes%20that%20are%20used%20in%20the%20message%20are%20included%20in%20the%20MQTT%20header.%20If%20a%20UAttributes%20field%20is%20not%20present%20in%20the%20header%2C%20than%20it%20is%20considered%20not%20used%20when%20recompiling%20the%20UAttributes

Reply:
As a check too before creating the user properties, you could validate the uAttributes from the uMessage with this [validator from up-java](https://github.com/eclipse-uprotocol/up-java/blob/main/src/main/java/org/eclipse/uprotocol/transport/validate/UAttributesValidator.java).

Reply:
If you look at the uAttributes spec you will see that not every field should be set depending on a message type and within those types not every value is mandatory. [uAttributes](https://github.com/eclipse-uprotocol/up-spec/blob/main/basics/uattributes.adoc). It is a good question on how to distinguish between the default value or a set value for uAttributes in java. @stevenhartley do you have any suggestions?

Reply:
discussed during the call today to address this